Update Get-LatestOSSupported.sh adding support for macOS Tahoe

HowardGMac opened the pull request:
Added support for macOS Tahoe (26.x)

Did check hardware regex against several models both capable and not capable of running macOS Tahoe, but not an exhaustive check.

HowardGMac committed:
Update Get-LatestOSSupported.sh for macOS Tahoe

First attempt at adding support for macOS Tahoe to script.

HowardGMac committed:
Update Get-LatestOSSupported.sh additional updates

dan-snelson commented:
Thanks, @HowardGMac; we're testing this in our Stage lane.

rougegoat commented:
Small tweak I'd make is swapping in this official URL on line 25.  Now that the macOS 26 is on the verge of releasing, the URL in there now redirects to that.
https://support.apple.com/en-us/120282

rougegoat commented:
Also just noticed this will get a false positive on the `macbookpro16,3`, which is the 2 port version.  The other three `macbookpro16,#` ones are supported by Tahoe, so not sure the best tweak to regex for it

rougegoat commented:
ok additional testing, this should work to get the last couple false positives out
`/^(MacPro[1-6],[0-9]|iMac([1-9]|1[0-9]),[0-9]|iMacPro1,1|Macmini[1-8],[0-9]|MacBookAir[1-9],[0-9]|MacBookPro((16,3)|([1-9]|1[0-5]),[0-9]))$`

HowardGMac committed:
Additional updates

Incorporated some additional updates to my pull request based on comment feedback

HowardGMac commented:
Incorporated the suggestions rougegoat made into my pull request. Updated the regex, updated the link for macOS Sequoia and added the place holder for Tahoe.

Angel038-2 commented:
Hi,
I use this script as extension atribute in jamf pro instance.
Field appears in computer informations, but return blank result
Testing script with code runner return good value
What could i do to solve this issue.?

MLBZ521 committed:
Bump version string

And add Tahoe URL

MLBZ521 commented:
Thank for you this PR.  Sorry it's taken me so long to merge it.

I just bump the version string and added the Tahoe URL before merging.

MLBZ521 commented:
@Angel038-2 I'm not sure why it wouldn't be working...  I would recommend running a verbose inventory on a device and see what is spit out.

Also, I would recommend opening an actual issue for this as this PR has been merged/closed.
